yapapi/runner/__init__.py

```diff
@@ -449,10 +448,13 @@ async def fill_work_q():
             ):
                 if datetime.now(timezone.utc) > self._expires:
                     raise TimeoutError(f"task timeout exceeded. timeout={self._conf.timeout}")
-
                 done, pending = await asyncio.wait(
                     services.union(workers), timeout=10, return_when=asyncio.FIRST_COMPLETED
                 )
+                for task in done:
+                    # if an exception occurred when a service task was running
+                    if task in services and not task.cancelled() and task.exception():
+                        raise Exception(task.exception())
```

Comment:
Why wrap `task.exception()` in another `Exception`?

Reply:
Because without this the type checker says the type is different (it can be `None`).

Reply:
We can also write `e: Exception = task.exception()`, because `task.exception()` is not None at this point.

Reply:
Yeah, I know why `mypy` complains, but wrapping it with `Exception` is not a correct solution (it would make catching and examining the original exception much harder), please use an `assert` or `typing.cast` as a hint for the typechecker (see e.g. https://mypy.readthedocs.io/en/stable/casts.html).